vector/src/main/java/im/vector/app/features/analytics/DecryptionFailureTracker.kt

```diff
@@ -136,20 +136,19 @@ class DecryptionFailureTracker @Inject constructor(
                     // for now we ignore events already reported even if displayed again?
                     .filter { alreadyReported.contains(it).not() }
                     .forEach { failedEventId ->
-                        analyticsTracker.capture(Error(failedEventId, Error.Domain.E2EE, aggregation.key))
+                        analyticsTracker.capture(Error(aggregation.key.first, Error.Domain.E2EE, aggregation.key.second))
                         alreadyReported.add(failedEventId)
                     }
         }
     }
 
-    private fun MXCryptoError.ErrorType.toAnalyticsErrorName(): Error.Name {
+    private fun MXCryptoError.ErrorType.toAnalyticsErrorName(): DetailedErrorName {
+        val detailed = "$name | mxc_crypto_error_type"
         return when (this) {
-            MXCryptoError.ErrorType.UNKNOWN_INBOUND_SESSION_ID -> Error.Name.OlmKeysNotSentError
-            MXCryptoError.ErrorType.OLM                        -> {
-                Error.Name.OlmUnspecifiedError
-            }
-            MXCryptoError.ErrorType.UNKNOWN_MESSAGE_INDEX      -> Error.Name.OlmIndexError
-            else                                               -> Error.Name.UnknownError
+            MXCryptoError.ErrorType.UNKNOWN_INBOUND_SESSION_ID -> Pair(detailed, Error.Name.OlmKeysNotSentError)
+            MXCryptoError.ErrorType.OLM                        -> Pair(detailed, Error.Name.OlmUnspecifiedError)
+            MXCryptoError.ErrorType.UNKNOWN_MESSAGE_INDEX      -> Pair(detailed, Error.Name.OlmIndexError)
```

Comment:
I think we can add MXCryptoError.ErrorType.KEYS_WITHHELD -> OlmKeysNotSentError
even if it could be a OlmIndexError

Reply:
Done, lets try that